framework/src/base/Attributes.C

```diff
@@ -309,38 +309,68 @@ AttribPreIC::isEqual(const Attribute & other) const
   return isMatch(other);
 }
 
-void
-AttribPreAux::initFrom(const MooseObject * /*obj*/)
-{
-}
 bool
 AttribPreAux::isMatch(const Attribute & other) const
 {
   auto a = dynamic_cast<const AttribPreAux *>(&other);
-  return a && (a->_val == _val);
+  if (!a || a->_vals.size() < 1)
+    return false;
+  auto cond = a->_vals[0];
+
+  for (auto val : _vals)
+    if (val == cond)
+      return true;
+  return false;
 }
 
 bool
 AttribPreAux::isEqual(const Attribute & other) const
 {
-  return isMatch(other);
+  auto a = dynamic_cast<const AttribPreAux *>(&other);
+  if (!a || a->_vals.size() != _vals.size())
+    return false;
+
+  for (size_t i = 0; i < a->_vals.size(); i++)
+    if (a->_vals[i] != _vals[i])
+      return false;
+  return true;
 }
 
 void
-AttribPostAux::initFrom(const MooseObject * /*obj*/)
+AttribPreAux::initFrom(const MooseObject * /*obj*/)
 {
 }
+
 bool
 AttribPostAux::isMatch(const Attribute & other) const
 {
   auto a = dynamic_cast<const AttribPostAux *>(&other);
-  return a && (a->_val == _val);
+  if (!a || a->_vals.size() < 1)
+    return false;
+  auto cond = a->_vals[0];
+
+  for (auto val : _vals)
+    if (val == cond)
+      return true;
+  return false;
 }
 
 bool
 AttribPostAux::isEqual(const Attribute & other) const
 {
-  return isMatch(other);
+  auto a = dynamic_cast<const AttribPostAux *>(&other);
+  if (!a || a->_vals.size() != _vals.size())
+    return false;
+
+  for (size_t i = 0; i < a->_vals.size(); i++)
+    if (a->_vals[i] != _vals[i])
+      return false;
```

Comment:
Same here with the sorting.

Reply:
I made `_vals` for `AttribPreAux` and `AttribPostAux` a `std::set` instead of a `std::vector`. I think this change addresses your comments about the order of _vals. Now `isEqual()` just checks if the sets are `==`.

I had to make other changes to get this to work. I think the reason that `AttribExecOns` used a `std::vector` for `_vals` and compared the order in `AttribExecOns::isEqual()` is because of the member function `hashfunc(_vals)`. The hash function specialization for std::vector just takes the elements of `_vals` sequentially and returns a hash value based on that input. So vectors in different orders will return different hash values. I made an additional specialization for std::set so that now `AttribPreAux::isEqual()` and `AttribPostAux::isEqual()` will return true when the sets are the same and also `hashfunc(_vals)` will return the same hash value. You can see this small code change in MooseHashing.h